xchange-core/src/main/java/org/knowm/xchange/instrument/SwapContract.java

```diff
@@ -0,0 +1,103 @@
+package org.knowm.xchange.instrument;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import java.io.Serializable;
+import org.knowm.xchange.currency.Currency;
+import org.knowm.xchange.currency.CurrencyPair;
+
+public class SwapContract extends Instrument implements Comparable<SwapContract>, Serializable {
+  /** */
+  private static final long serialVersionUID = -8400518639933611025L;
+
+  public final String maturity;
+  public final CurrencyPair underlier;
+
+  /**
+   * Full constructor In general the prompt is an approximation of the swap maturity you're wanting
+   * to buy/sell.
+   *
+   * @param prompt The approximated maturity of the swap contract
+   */
+  public SwapContract(CurrencyPair underlier, String maturity) {
+    this.underlier = underlier;
+    this.maturity = maturity;
+  }
+
+  @JsonCreator
+  public SwapContract(String swapContract) {
+
+    String[] parts = swapContract.split("/");
```

Comment:
how will you InstrumentDeserializer distinguish a `SwapContract` from a `FutureContract` if they have the same amount of `/` ?